source/extensions/common/wasm/v8/v8.cc

```diff
@@ -0,0 +1,590 @@
+  _REGISTER_HOST_FUNCTION(WasmCallback2Int);
+  _REGISTER_HOST_FUNCTION(WasmCallback3Int);
+  _REGISTER_HOST_FUNCTION(WasmCallback4Int);
+  _REGISTER_HOST_FUNCTION(WasmCallback5Int);
+  _REGISTER_HOST_FUNCTION(WasmCallback6Int);
+  _REGISTER_HOST_FUNCTION(WasmCallback7Int);
+  _REGISTER_HOST_FUNCTION(WasmCallback8Int);
+  _REGISTER_HOST_FUNCTION(WasmCallback9Int);
+  _REGISTER_HOST_FUNCTION(WasmCallback_Zjl);
+  _REGISTER_HOST_FUNCTION(WasmCallback_Zjm);
+  _REGISTER_HOST_FUNCTION(WasmCallback_mj);
+  _REGISTER_HOST_FUNCTION(WasmCallback_mjj);
+#undef _REGISTER_HOST_FUNCTION
+
+#define _GET_MODULE_FUNCTION(_type)                                                                \
+  void getFunction(absl::string_view functionName, _type* f) override {                            \
+    getModuleFunctionImpl(functionName, f);                                                        \
+  };
+  _GET_MODULE_FUNCTION(WasmCall0Void);
+  _GET_MODULE_FUNCTION(WasmCall1Void);
+  _GET_MODULE_FUNCTION(WasmCall2Void);
+  _GET_MODULE_FUNCTION(WasmCall3Void);
+  _GET_MODULE_FUNCTION(WasmCall4Void);
+  _GET_MODULE_FUNCTION(WasmCall5Void);
+  _GET_MODULE_FUNCTION(WasmCall8Void);
+  _GET_MODULE_FUNCTION(WasmCall1Int);
+  _GET_MODULE_FUNCTION(WasmCall3Int);
+#undef _GET_MODULE_FUNCTION
+
+private:
+  void callModuleFunction(Context* context, absl::string_view functionName, wasm::Val args[],
+                          wasm::Val results[]);
+
+  template <typename T>
+  std::unique_ptr<Global<T>> registerHostGlobalImpl(absl::string_view moduleName,
+                                                    absl::string_view name, T initialValue);
+  template <typename... Args>
+  void registerHostFunctionImpl(absl::string_view moduleName, absl::string_view functionName,
+                                void (*function)(void*, Args...));
+  template <typename R, typename... Args>
+  void registerHostFunctionImpl(absl::string_view moduleName, absl::string_view functionName,
+                                R (*function)(void*, Args...));
+  template <typename... Args>
+  void getModuleFunctionImpl(absl::string_view functionName,
+                             std::function<void(Context*, Args...)>* function);
+  template <typename R, typename... Args>
+  void getModuleFunctionImpl(absl::string_view functionName,
+                             std::function<R(Context*, Args...)>* function);
+
+  wasm::vec<byte_t> source_ = wasm::vec<byte_t>::invalid();
+  wasm::own<wasm::Store*> store_;
+  wasm::own<wasm::Module*> module_;
+  wasm::own<wasm::Instance*> instance_;
+  wasm::own<wasm::Memory*> memory_;
+  wasm::own<wasm::Table*> table_;
+
+  absl::flat_hash_map<std::string, wasm::own<wasm::Global*>> host_globals_;
+  absl::flat_hash_map<std::string, wasm::own<wasm::Func*>> host_functions_;
+  absl::flat_hash_map<std::string, wasm::own<wasm::Func*>> module_functions_;
+  bool module_needs_emscripten_{};
+};
+
+// Helper functions.
+
+const char* printValKind(wasm::ValKind kind) {
```

Comment:
Hmm, I guess this is not supposed be prepared for ADL, maybe you wanna declare it static, or put in anonymous namespace

Reply:
Done (declared static).